### ISSUE TITLE
Error when `Protocol` method has only docstring and no implementation

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#3626]
**Describe the bug**
When I create a `Protocol` method that has only a docstring and no implementation I get the following warning.
```python
class Labels(Protocol):
    """User friendly information"""

    def name(self) -> str:  # Function with declared type of "str" must return value. Type "None" cannot be assigned to type "str"
        """The name"""
```

 Adding `...` gets rid of the error but should be unnecessary and causes a [Pylint warning](https://pylint.pycqa.org/en/latest/user_guide/messages/warning/unnecessary-ellipsis.html) instead.

I expected to be able to just have the docstring without any error.

**VS Code extension or command-line**
Pylance language server 2022.6.30 (pyright 212d1465)

**Comment by erictraut** [comment]
This is intended. Protocol classes can include method implementations, but they can also include method "prototypes" that must be implemented by a class that derives from the protocol class. Pyright must distinguish between these two cases. If it sees an implementation of `...`, it assumes that the method is a prototype. If it sees something other than that (including a `pass` or a docstring), it assumes that the method is intended to be an implementation, and it is type checked accordingly.

**Comment by parched** [comment]
Ok, I can understand the need to distinguish between the two, but surely docstring only indicates a prototype? Is there any precedent that it must have `...` to be a prototype? If there is, I'll raise a bug with Pylint instead. FWIW, Mypy is happy with docstring only.

**Comment by erictraut** [comment]
What is the objection to adding `...` to indicate that the body of the method is intended to be filled in by a subclass? That seems much less ambiguous than relying solely on the presence of a docstring.

I'd prefer not to change this because it would create an inconsistency in pyright's behavior.

This sounds like pylint being a bit too opinionated (or having an incorrect opinion) in this case.
